Feature/228(로그인하지 않은 사용자 페이지 이동 제한)

han0224 opened the pull request:
## #️⃣ 연관된 이슈 번호

> 연관된 이슈 번호를 모두 작성해주세요

Close #228 

## ✍🏻 작업 내용

> 이번 PR에서 작업한 내용을 간략히 설명해주세요(이미지 첨부 가능)

- 로그인하지 않은 유저: localStorage에 토큰이 저장되어 있지 않은 유저
- 로그인하지 않은 유저가 메인 페이지외 다른 페이지를 이동할려할시 경고창과 메인페이지로 강제 이동

- 로그인하지 않은 관리자: pinia에 토큰이 저장되어 있지 않은 관리자
- 관리자가 로그인 하지 않고 로그인 외 다른 관리자 페이지에 접근할 시 경고창과 관리자 로그인 페이지로 강제 이동

## 💬 리뷰 요구사항(선택)

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요

현재 코드 기준으로 관리자는 새로고침하면 로그인이 바로 풀리는 방식인데 계속 이렇게 진행해도 괜찮을까요?

han0224 committed:
fix: 모든 상품 조회 api 반환값 수정

han0224 committed:
feat: 로그인하지 않은 유저는 메인페이지 제외 접근불가능

- 메인페이지에 meta:{every:true}} 추가
- 이동할려는 페이지에 meta.every가 없고 로그인하지 않은 유저일 경우 alert 창과 메인페이지로 이동
- 로그인하지 않은 유저는 pinia에 데이터가 저장되어 있지 않은 유저로 정의

han0224 committed:
refactor: admin 페이지는 경로 접근시 import하도록 변경

han0224 committed:
fix: 관리자 로그인 후 새로고침 삭제

han0224 committed:
fix: 로그인하지 않은 유저 재정의

- pinia에 저장이 아니라 localStorage에 token에 저장이 되어있어야 로그인한 유저로 정의
- 로그인한 사용자가 url을 직접입력하고 페이지를 넘어가면 pinia에 저장된 데이터가 사라짐
- 그렇기에 url을 직접입력해도 데이터가 사라지지않는 localStorage 에 토큰이 저장된 유저를 로그인한 유저로 정의

han0224 committed:
feat: error 메세지추가

han0224 committed:
refactor: 문자열 분리

devFancy commented on frontend/src/router/index.ts:
```diff
@@ -61,33 +60,88 @@ const router = createRouter({
+  }
+  if (to.meta.type === 'admin' && !adminStore.accessToken) {
+    alert(ERROR_MSG.NEED_LOGIN);
+    goPageWithReload('admin');
```
`goPageWithReload`는  로그인 페이지로 이동하는 건가요? 맞다면 해당 메서드 명을 직관적으로 이해되기 쉽게 수정하면 좋을 것 같아요~

han0224 replied:
해당 메소드는 새로고침해서 페이지를 이동하는거입니다! 관리자 로그인 페이지 url이 admin 이여서 인자값으로 admin으로 보내준것입니다!
해당 메서드 명을 직접 변경하는 것보다 url을 변경하는게 나을것같네요. 이후 페이지별 url을 정확하게 정하고 나서 변경하는게 좋을 것 같아요!

devFancy replied:
아 그렇군요! 알겠습니다~

devFancy reviewed (approved):
수고하셨습니다~

devFancy reviewed (approved):
수고하셨습니다~

han0224 committed:
fix: api 요청 실패메세지

error.message 는 axios에서 제공해주는 메세지
error.response.data.message 에 서버에서받은 에러 메세지 존재
